flexiblepricing/management/commands/configure_for_dedp.py

```diff
@@ -0,0 +1,195 @@
+from datetime import date, datetime
+from django.core.management import BaseCommand
+from django.contrib.contenttypes.models import ContentType
+from django.conf import settings
+import pytz
+
+from courses.models import Program
+from flexiblepricing.models import FlexiblePriceTier
+from ecommerce.models import Discount
+from ecommerce.constants import REDEMPTION_TYPE_UNLIMITED
+
+
+class Command(BaseCommand):
+    """
+    Sets up some basic records for DEDP Financial Assistance
+    """
+
+    help = "Sets up some basic records for DEDP Financial Assistance"
+    PROGRAM_READABLE_ID = "program-v1:MITx+DEDP"
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        current_year = date.today().year
+        last_year = datetime(
+            current_year - 1, 1, 1, tzinfo=pytz.timezone(settings.TIME_ZONE)
+        )
+        discounts_and_tiers = [
+            {
+                "tier": {"threshold": 0},
+                "discount": {
+                    "discount_code": f"DEDP-fa-tier1-{current_year}",
+                    "discount_type": "dollars-off",
+                    "amount": 750,
+                },
+            },
+            {
+                "tier": {"threshold": 25000},
+                "discount": {
+                    "discount_code": f"DEDP-fa-tier2-{current_year}",
+                    "discount_type": "dollars-off",
+                    "amount": 650,
+                },
+            },
+            {
+                "tier": {"threshold": 50000},
+                "discount": {
+                    "discount_code": f"DEDP-fa-tier3-{current_year}",
+                    "discount_type": "dollars-off",
+                    "amount": 500,
+                },
+            },
+            {
+                "tier": {"threshold": 75000},
+                "discount": {
+                    "discount_code": f"DEDP-fa-tier4-{current_year}",
+                    "discount_type": "percent-off",
+                    "amount": 0,
+                },
+            },
+        ]
+        content_type = ContentType.objects.filter(
+            app_label="courses", model="program"
+        ).first()
+
+        # Step one: get the DEDP program
+        self.stdout.write(self.style.NOTICE("Setting up the program..."))
+        (program, created) = Program.objects.update_or_create(
+            readable_id=self.PROGRAM_READABLE_ID,
+            defaults={"title": "Data, Economics and Development Policy", "live": True},
+        )
+        if created:
+            self.stdout.write(self.style.NOTICE(f"Created new program {program.id}"))
+        else:
+            self.stdout.write(self.style.NOTICE(f"Using existing program {program.id}"))
+
+        # Step two: get existing discounts
+        discounts = Discount.objects.filter(
+            discount_code__startswith="DEDP-fa-tier"
+        ).all()
+        self.stdout.write(
+            self.style.NOTICE(f"{len(discounts)} existing discounts for {current_year}")
```

Comment:
Is this intended to assume existing discounts are for the current year?

Reply:
ignore me. I misread this text. I think my question was on the different line 135-138